docs/release-branch-backporting.md

```diff
@@ -0,0 +1,28 @@
+# Release Branch Backporting Policy
+
+Bug fixes are eligible to be backported from the master branch to any previous
+release branch. The following criteria must be met before a backport can be
+considered. It is the reviewer's and approver's responsibility to uphold this
+policy.
+
+- **Bug Fix Only:** The backport must be a bug fix and the bug fix must be
+first merged into the master branch. The only exception is when a bug only
+exists in a stable branch and does not exist in the master branch.
```

Comment:
Can we be explicit about preferring that a url describing the bug/issue is included in the pr text, and preferably also in the commit message?

Reply:
I'm not sure about the commit messages. These are (most of the time) being backported automatically with a label `/cherrypick [release]`
If we want bugs to be added to commit messages we probably need to automate this process.
@rmohr what do you think?

Reply:
> Can we be explicit about preferring that a url describing the bug/issue is included in the pr text

sure, doesn't the "Detailed Description" section in this doc cover this though? 

```
- **Detailed Description** The backport pull request's description must either
contain a detailed explanation for what the bug fix addresses or contain a
link to such a explanation present in another PR/Issue.
```

> preferably also in the commit message?

This will be harder to enforce. Often bug fixes are split across multiple commits.

Reply:
Maybe it would make sense to just mention the issue id or bug url in the release note itself? This would then automatically and directly end up in the release description? Or would that be too much?

Reply:
I'd prefer a full URL. I'd assume bug trackers of multiple vendors might be involved, so better to be explicit.

Reply:
Please note the "Detailed Description" section does a nice job of answering this IMO. A reference to a bug/issue is not explicitly required in a sufficient explanation is given directly.

Reply:
Perhaps, we should stop using bugs altogether and only use issues.
The in the issue we can specify bugs or any other public issue tracker from any company.

Reply:
I would recommend any company to use its internal tracking URL if it wants to, right in the commit message. It helps a lot, as the URL sticks with the commit as it is backported, and no harm is done upstream if another URL is added.

The stable branch maintainer *may* say: "hey, this PR is frightening, I want to see a github issue before I'm backporting it" but I would not force all backports to have an github issue. Any sufficiently-clear freely-readable maintainer-satisfying explanation is good for me.

Reply:
I don't see neither Kubernetes nor Openstack Nova uses company-specific bug trackers. I can't tell how healthy will be for the project long term.

Reply:
Maybe I wasn't clear. I'm not suggesting KubeVirt to use private bugtrackers. I'm just saying that we can tolerate any kind of private reference tucked in the commit message.

Reply:
Isn't it good enough, like david said?

> > Can we be explicit about preferring that a url describing the bug/issue is included in the pr text
> 
> sure, doesn't the "Detailed Description" section in this doc cover this though?

Reply:
The main thing I care about is that backports PRs provide a clear case for why the backport is necessary. That gives us two things.

1. As a reviewer/approver  i can make a judgement call for whether the PR is a bug fix and what functionality it impacts.
2. Anyone can retroactively trace back the commit to the PR to discover why the bug fix was introduced.

My primary goal here is to ensure that both the defense and reasoning for a backport is transparent. This avoids the cycle where a couple of approvers get together out of band and decide something must be backported without documenting any sort of hint for why the backport is necessary.

I'm probably the primary offender here. This is to protect the project from me :D

So, from my standpoint, the minimum is to have all this info in the PR description. Having full info in a commit message is nice, but like i said previously, difficult to enforce (especially with our cherry-pick plugin).  I don't think it's reasonable to ask someone to cherry-pick a PR, then modify the commit messages for whatever is cherry-picked. Instead they should supplement the PR's description with more info if needed for tracking why the PR is being backported.

Reply:
> Isn't it good enough, like david said?
> 
> > > Can we be explicit about preferring that a url describing the bug/issue is included in the pr text
> > 
> > 
> > sure, doesn't the "Detailed Description" section in this doc cover this though?

It is certainly good enough.